jnigen/lib/src/bindings/dart_generator.dart

```diff
@@ -1131,3 +1225,137 @@ class _JValueWrapper extends TypeVisitor<String> {
     return '$_jni.JValue${node.name.capitalize()}($param)';
   }
 }
+
+/// A [StringBuffer] that prepends instead of appending on write.
```

Comment:
This feels a bit of a wacky abstraction. Prepending and adding parenthesis feel like two orthogonal concerns.

I gather string concatenation or normal StringBuffer didn't suffice.

Should this be a reusable abstraction in its own file? In utils? Or are we only going to is it in this dart_generator?

Reply:
Both of them do suffice but they're going to be slower. For one, we have to prepend instead of appending, doing it using a normal `StringBuffer` would make it `O(n)`, and we would want to keep the count of the closing parenthesis to be able to open them. 

This is not going to be reusable in any other context. Just for this place.